sos/plugins/validations.py

```diff
@@ -0,0 +1,44 @@
+# Copyright (C) 2020 Red Hat, Inc., Cedric Jeanneret <cjeanner@redhat.com>
+
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.plugins import Plugin, RedHatPlugin
+import re
+
+
+# Notes:
+# - The Validation Framework is, for now, linked to openstack and tripleo
+# - Since the intent is to open it to other product (we can validate anything)
+#   this plugin has a generic name.
+# - The Framework is targeted at Red Hat products, at least for now.
+class Validations(Plugin, RedHatPlugin):
+    """Fetch run logs provided by the Validation Framework
+    """
+
+    plugin_name = 'validations'
+    profiles = ('openstack', 'openstack_controller', 'openstack_compute')
+    packages = ('tripleo-validations',)
+
+    def setup(self):
+        self.add_copy_spec([
+            '/var/log/validations/',
+        ])
+
+    def postproc(self):
+        # Use a generic match in order to clean things up.
+        # It is not expected to get any secrets in here, but we'd better
+        # ensure it's clean.
+        secrets = r'(".*(key|password|pass|secret|database_connection))' \
+                  r'([":\s]+)(.*[^"])([",]+)'
+        rgxp = re.compile(secrets, re.IGNORECASE)
```

Comment:
`do_path_regex_sub` already does this compilation step, so no need to do it here and we can drop the `re` import for this plugin.

Reply:
ah, I didn't know about that one, and it wasn't pointed when I created another plugin lately. Lemme correct it.